### PR TITLE
Implement ListEvents Runtime Interface to Support Cross-Cluster Events

### LINKED ISSUE #563: Implement `ListEvents` Runtime Interface to Support Cross-Cluster Events
# Summary

This proposal aims to implement the Runtime interface to support cross-cluster events, enhancing observability and debug-ability (see also https://github.com/ironcore-dev/ironcore/issues/1070).

## Basic example

Events of the `ceph-provider` can be stored in-memory until they are fetched by the `poollet`. To avoid exceeding memory limits, events should be garbage collected after a defined TTL and the event store should evict the oldest events in case of reaching the max capacity.

### PR DESCRIPTION
**Pull request by ushabelgur**
# Proposed Changes

- Add ListEvents implementation to volume provider
- Add event recorder store implementation 

Fixes #563

### CODE REVIEW

File: internal/controllers/image_controller.go [hunk 1]
```diff
@@ -235,7 +242,7 @@ type fetchAuthResponse struct {
 	Key string `json:"key"`
 }
 
-func (r *ImageReconciler) fetchAuth(ctx context.Context, log logr.Logger) (string, string, error) {
```
Comment (lukasfrank) [inline internal/controllers/image_controller.go:238]: Can we please keep the `ctx`? Currently, a lot of the `ceph` commands do not support a `ctx`, however libraries tend to add a `ctx` from time to time. If we run into issues we can also implement our cancelation logic.

**Review by lukasfrank — CHANGES_REQUESTED** [review]

File: internal/controllers/image_controller.go (hunk 1 above)
Comment (ushabelgur) [reply internal/controllers/image_controller.go]: reverted back changes

**Comment by lukasfrank** [comment]
Since `reconcileSnapshot` influences the processing time of the `image` provisioning it would be great to add more context by writing events there.

**Comment by ushabelgur** [comment]
> Since `reconcileSnapshot` influences the processing time of the `image` provisioning it would be great to add more context by writing events there.

Volume metadata is attached only with image, its not being propagated to snapshots currently. If i generate events for snapshot with this volume metadata information, it will not be useful in propagating event. Will try to check if i can populate this info in snapshot.

**Comment by lukasfrank** [comment]
> > Since `reconcileSnapshot` influences the processing time of the `image` provisioning it would be great to add more context by writing events there.
> 
> Volume metadata is attached only with image, its not being propagated to snapshots currently. If i generate events for snapshot without this volume metadata information, it will not be useful in propagating event. Will try to check if i can populate this info in snapshot.

Can't we say here, that the snapshot was not present and therefore we gonna create it. I think this is still useful information and we also have the image.

**Review by lukasfrank — APPROVED** [review]

**Pull request merged**